src/ir/expr.rs

```diff
@@ -14,10 +14,20 @@ pub enum Expr {
     },
     Fn {
         op: ast::UnFn,
-        args: Box<[ExprIdx]>,
+        args: Box<Vec<ExprIdx>>,
     },
 }
 
+impl Expr {
+    pub fn as_concrete(&self) -> Option<u64> {
```

Comment:
There is another `as_concrete` method on `ExprIdx`. In general, prefer defining methods on `ExprIdx` instead of `Expr` because most of the time, user code will have access to the former